src/test/java/org/springframework/samples/petclinic/owner/VisitControllerTests.java

```diff
@@ -80,4 +83,26 @@ public void testProcessNewVisitFormHasErrors() throws Exception {
             .andExpect(view().name("pets/createOrUpdateVisitForm"));
         verify(pets).findById(anyInt()); //Ensure that a pet gets loaded per visit
     }
+    
+    @Test
+    public void testMockVisitRepository() {
+    	//Use the fake repository to break dependency of visit controller on visit repository
+    	FakeVisitRepository fakeVisits = new FakeVisitRepository();
+    	
+    	//Creation of visit controller with fake visit repository and mocked pets by mockMVC
+    	VisitController visitController = new VisitController(fakeVisits, pets);
+    	
+    	//Creation of visit to add to visits repo and mocked result to invoke a visit controller method that involves
+    	//the use of the visits repo
+    	Visit visit = mock(Visit.class);
+    	when(visit.getId()).thenReturn(22);
+    	BindingResult result = mock(BindingResult.class);
+    	
+    	//internally this will use the fake visit repository's save method; other methods rely on the mocked pets repository
+    	visitController.processNewVisitForm(visit, result);
+    	
+    	//Check that save method was exectued by checking that the id of the visit has now been stored
+    	assertEquals(visit.getId().intValue(), fakeVisits.getVisitId(visit));
```

Comment:
Instead of using `getVisitId` maybe use `findByPetId` and since it returns an `ArrayList`, get the first result from that for the assertion

Reply:
Good catch, will address this in next commit.